### PR TITLE
Chore/178726224 run scxa tests in gradle container

### PR DESCRIPTION
**Pull request by upendrakumbham**
This PR contains the required changes to run unit, integration, and web integration tests in the Gradle container and split the docker-compose files into multiple files based on dependencies to reuse configuration.

Now application, tests are reusing configuration.

Please read this section:` Run Tests using Gradle container` before moving further to test these changes.
[README](https://github.com/ebi-gene-expression-group/atlas-web-single-cell/blob/chore/178726224-run-scxa-tests-in-gradle-container/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -206,15 +206,48 @@ Read the important message after you run `scxa-solrlcoud-bootstrap`:
 > takes a few extra minutes: the exception is thrown before the process has completed.
 > The best option is to manually build and supervise this step.
 >
-> Run the following command (don’t worry if the request returns a 500 error after a while):
+> Run the following commands one by one(don’t worry if the request returns a 500 error after a while):
 > 
 > `docker exec -i scxa-solrcloud-1 curl 'http://localhost:8983/solr/bioentities-v1/suggest?suggest.build=true&suggest.dictionary=propertySuggesterNoHighlight'`
+
+> `docker exec -i scxa-solrcloud-1 curl 'http://localhost:8983/solr/bioentities-v1/suggest?suggest.build=true&suggest.dictionary=bioentitySuggester'`
```
Comment (alfonsomunozpomer) [inline README.md:213]: I just realised that we can run all the suggesters in a single command. I’ve tested it and it works:
```bash
docker exec -i scxa-solrcloud-1 curl 'http://localhost:8983/solr/bioentities-v1/suggest?suggest.build=true&suggest.dictionary=bioentitySuggester&suggest.dictionary=propertySuggesterNoHighlight'

```

Comment (alfonsomunozpomer) [inline README.md:213]: I just realised that we can run all the suggesters in a single command. I’ve tested it and it works:
```bash
docker exec -i scxa-solrcloud-1 curl 'http://localhost:8983/solr/bioentities-v1/suggest?suggest.build=true&suggest.dictionary=bioentitySuggester&suggest.dictionary=propertySuggesterNoHighlight'
```

File: README.md [hunk 2]
```diff
@@ -206,15 +206,48 @@ Read the important message after you run `scxa-solrlcoud-bootstrap`:
...
 >
 > While the command above is running, monitor the size of the suggester directory size:
 > 
 > `docker exec -it scxa-solrcloud-1 bash -c 'watch du -sc server/solr/bioentities-v1*/data/*'`
 > `docker exec -it scxa-solrcloud-2 bash -c 'watch du -sc server/solr/bioentities-v1*/data/*'`
 >
-> The suggester in all shards will be built when the propertySuggester directory size stabilises. 
-> Run the above procedure for `bioentitySuggester` as well.
- 
+> The suggester in all shards will be built when the propertySuggester directory size stabilises.
```
Comment (alfonsomunozpomer) [inline README.md:220]: We can generalise this comment for all suggesters then:
> Both suggesters in every shard will be built when the suggester directory size stabilises.

**Review by alfonsomunozpomer — CHANGES_REQUESTED** [review]
Besides the comment about the suggesters, the Docker Compose files should be restructured. It’s a good opportunity to split the our `docker-compose.yml` into the following:
- SolrCloud cluster: Solr + ZooKeeper containers
- Postgres: PostgreSQL + Flyway containers
- Tomcat
- Gradle

Have a look at [Multiple Compose files](https://docs.docker.com/compose/extends/#multiple-compose-files) and [Extending services](https://docs.docker.com/compose/extends/) in the Docker Compose documentation.

File: README.md [hunk 3]
```diff
@@ -206,15 +206,48 @@ Read the important message after you run `scxa-solrlcoud-bootstrap`:
...
+
+In the `atlas-web-single-cell/docker` directory run the following:
+
+```bash
+ATLAS_DATA_PATH=/path/to/sc/atlas/data \
+GRADLE_HOST=scxa-gradle\
+PWD=/path/to/sc/project \
+ATLAS_MAVEN_CACHE=/path/to/maven/cache \
+ATLAS_GRADLE_CACHE=/path/to/gradle/cache \
+docker-compose -f docker-compose-test.yml up
```
Comment (lingyun1010) [inline README.md:236]: I think you mean `docker-compose-gradle` instead of `docker-compose-test` here.

Comment (upendrakumbham) [reply README.md:236]: yes, sorry, copy paster error :)

File: README.md (hunk 1 above)
Comment (upendrakumbham) [reply README.md:213]: Nice

File: README.md (hunk 2 above)
Comment (upendrakumbham) [reply README.md:220]: Sure

Comment (upendrakumbham) [reply README.md]: Fixed

File: README.md (hunk 3 above)
Comment (upendrakumbham) [reply README.md]: Fixed

**Review by lingyun1010 — DISMISSED** [review]
Works fine for me!

**Comment by pcm32** [comment]
If this is working for both @lingyun1010 and @upendrakumbham, and given that in my opinion the changes requested by @alfonsomunozpomer have been satisfied, I'm happy to merge this as it is. I will give it a couple of hours, but if I don't hear any objections, I will merge as it is.

**Review by alfonsomunozpomer — APPROVED** [review]
All good, some tests are failing but we’ll fix them later.

**Review by alfonsomunozpomer — APPROVED** [review]
See my earlier review. :smile:

**Pull request merged**